packages/client/src/clients/in-app-message/channel-client.ts

```diff
@@ -0,0 +1,18 @@
+import Knock from "../../knock";
+
+import { InAppStore, createStore } from "./store";
+
+/**
+ * Manages the configuration for an in app channel.
+ * Stores all fetched messages to support optimistic updates.
+ */
+export class InAppChannelClient {
+  public store: InAppStore;
+
+  constructor(
+    readonly knock: Knock,
+    readonly channelId: string,
```

Comment:
It feels like there's also likely other shared/default options that can be passed here that can be shared across all message clients (tenant is a good example)

Reply:
Sounds good to me! So we'd cascade down options like ChannelClient > MessageClient > Fetch for JS. In React, that'd be KnockInAppChannelProvider > useInAppMessage(s) > messageClient.fetch (if called directly)